controllers/pvc-mutator/controller.go

```diff
@@ -66,3 +76,38 @@ func (c *Controller) Handle(ctx context.Context, req admission.Request) admissio
 
 	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPVC)
 }
+
+func (c *Controller) capacityCheck(ctx context.Context, pvc *corev1.PersistentVolumeClaim) error {
+	// Find StorageClass of PVC.
+	storageClass, err := provisioner.StorageClassForPVC(c.Client, pvc)
+	if err != nil {
+		return errors.Wrap(err, "failed to check pvc provisioner")
+	}
+
+	// Skip mutation if the PVC will not be provisioned by StorageOS.
+	if provisioned := provisioner.IsProvisionedStorageClass(storageClass, provisioner.DriverName); !provisioned {
+		c.log.V(4).Info("pvc will not be provisioned by StorageOS, skipping")
+		return nil
+	}
+
+	// Get int64 representation of requested storage from the pvc.
+	var storageRequested int64
+	var ok bool
+	if storageRequested, ok = pvc.Spec.Resources.Requests.Storage().AsInt64(); !ok {
+		if storageRequested, ok = pvc.Spec.Resources.Requests.Storage().ToDec().AsInt64(); !ok {
+			return fmt.Errorf("failed to get int64 representatoin of requested storage")
+		}
+	}
+
+	// Get licence from control plane.
+	licence, err := c.licence.GetLicence(ctx)
```

Comment:
Does it make sense to cache the result for a time period? I mean usually licence doesn't change in a minute bases.